### ISSUE TITLE
[Deque Analysis] Login module's "serious" accessibility issues

### ISSUE THREAD

**Issue by tdonohue** [DSpace/dspace-angular#1184]
> Serious: This issue results in serious barriers for people with disabilities, and will partially prevent them from accessing fundamental features or content. People relying on assistive technologies will experience significant frustration as a result. Issues falling under this category are major problems, and remediation should be a priority. 

## Deque Analysis Summary

A total of 3 serious issues exist (not including color contrast which are handled in other tickets) in our login module (available in the upper right on each page).

1. ~~(ID: 469328) "Label is not persistent" on the email/password fields. "placeholder is being used as the only visual label for a text field. The placeholder label is not visually available after data is entered in the text field."~~
    * ~~Recommended Fix: "Labels MUST be visible. Ensure that the label is programmatically associated with the form fields via the label/for technique or ARIA attributes."~~
    * FIXED ALREADY VIA Aria attributes (verified as of 7.5, though uncertain when it was actually fixed)
2. (ID 469335) "Identify Input Purpose Not Accurate." on the email/password fields. "Autocomplete attribute to identify the purpose of this form element is not accurate." (Currently both have `autocomplete="off"`)
    * Recommended fix: Fix this issue by using the autocomplete attribute with the appropriate value from HTML5.2 on the text input element. The list of 53 defined input purposes for UI components can be found here -
https://www.w3.org/TR/WCAG21/#input-purposes  (For our purposes, they likely should be set to "username" and "current-password")
3. (ID 469343) ~~"Text spacing content cut off." on the "Have you forgotten your password?" link.~~
    * ~~Recommended fix: Fix this issue by allowing users to override text spacing styles by providing flexibility in the height and width of content containers. Some common techniques include:~~
        * ~~Size containers to a have a value greater than the default width of the text. Small areas of text (such as navigation bars) need to have sufficient space within each box so that words and characters do not expand out of their containers.~~
        * ~~Use the inline-block value of the CSS display property and do not set negative margins to help allow for spacing overrides.~~
        * ~~Do not set the CSS height property, it can help ensure paragraphs are able to expand.~~
        * ~~See also https://www.w3.org/WAI/WCAG21/Understanding/text-spacing~~
        * NO LONGER REPRODUCIBLE IN 7.6.1.  Unsure when fixed.
4. (ID 469229) ~~"id attribute value must be unique". Document has multiple static elements with the same id attribute: `loginDropdownMenu`~~
    * ~~NOTE: Rated a "minor" issue, not "serious", but it may be simple to fix. If it's not simple, it should be delayed & moved to a different ticket.~~
    * FIXED BY #2683

Full list of issues is viewable at (requires login): https://axeauditor.dequecloud.com/test-run/0856438a-a19a-11eb-bc31-b7d5be387c86/issues?sortField=ordinal&sortDir=asc&filter%5Bseverity%5D=3&filter%5Btype%5D=issue&filter%5Bpage_number%5D=8
